src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -5452,7 +5459,7 @@ private static MethodHandle dropArgumentsToMatch(MethodHandle target, int skip,
     public static MethodHandle dropArgumentsToMatch(MethodHandle target, int skip, List<Class<?>> newTypes, int pos) {
         Objects.requireNonNull(target);
         Objects.requireNonNull(newTypes);
-        return dropArgumentsToMatch(target, skip, newTypes, pos, false);
+        return dropArgumentsToMatch(target, skip, newTypes.toArray(new Class<?>[0]).clone(), pos, false);
```

Comment:
Do we really need to clone an array returned from `List.toArray()`? As far as I know from the JavaDoc of `List` if the passed array is not long enough to include all the items then the new array must be allocated. Here we always pass empty arrays, so the new ones are returned from `toArray()` method and we don't need `clone()`, right?

Reply:
The clone is needed - as the `List<Class<?>>` may be a custom implementation of list - so you cannot make any assumption on the concrete implementation of `toArray`.

Reply:
But we can make an assumption on the behavior of particular method and the JavaDoc is clear about it. If a user passes an incorrect implementation of the `List` interface then it's their own responsibility.

Reply:
We can't rule out malicious code holding the array reference it returns in toArray and updates that array to cause damage.